cmd/vela-server/schedule.go

```diff
@@ -117,15 +116,25 @@ func processSchedules(ctx context.Context, start time.Time, compiler compiler.En
 		// send API call to update schedule for ensuring scheduled_at field is set
 		_, err = database.UpdateSchedule(ctx, schedule, false)
 		if err != nil {
-			logrus.WithError(err).Warnf("%s %s", scheduleErr, schedule.GetName())
+			handleError(ctx, database, err, schedule)
 
 			continue
 		}
 
 		// process the schedule and trigger a new build
 		err = processSchedule(ctx, schedule, compiler, database, metadata, queue, scm, allowList)
 		if err != nil {
-			logrus.WithError(err).Warnf("%s %s", scheduleErr, schedule.GetName())
+			handleError(ctx, database, err, schedule)
+
+			continue
+		}
+
+		schedule.SetError("")
+
+		// send API call to update schedule with the error message field cleared
+		_, err = database.UpdateSchedule(ctx, schedule, true)
+		if err != nil {
+			handleError(ctx, database, err, schedule)
```

Comment:
i don't think we want to keep updating schedules (read: make database calls) on a happy path to update error field. how about wrapping this in a conditional to only execute if the error field is non-empty on a given schedule?

Reply:
Yeah that makes sense. I will update that.